vaadin-combo-box-flow-demo/src/main/java/com/vaadin/flow/component/combobox/demo/ComboBoxView.java

```diff
@@ -455,6 +460,32 @@ private void filteringAndSortingWithDataView() {
                 personAgeFilter, sortPersons);
     }
 
+    private void itemCountChangeNotificationWithDataView() {
+        // begin-source-example
+        // source-example-heading: Notification of Items Count Change
+        ComboBox<String> comboBox = new ComboBox<>("Fruits");
+        comboBox.setPlaceholder("Enter a filter");
+
+        ComboBoxListDataView<String> dataView = comboBox.setItems("Apple",
+                "Orange", "Banana", "Kiwifruit", "Grapefruit");
+
+        Span fruitCountSpan = new Span(
+                "Fruits count: " + dataView.getItemCount());
+
+        // Client filter usually changes the displayed items count in combo
+        // box. With data view, one can subscribe to combo box's items count
+        // change events and get a notifications on server side.
+        dataView.addItemCountChangeListener(event -> fruitCountSpan
+                .setText("Fruits count: " + event.getItemCount()));
```

Comment:
I don't think this "valid" - it is not the fruit count but that the "matches count" when filtering. And that was what I meant with this example, that it would just show the number when filtered. And that number doesn't make any sense here with just so few items.

Maybe the idea for the demo was not that good after all - or at least like this it is just confusing. Not sure if it would just be about showing e.g. `(123)`  (all items) first and then when filtering it would show `(32/123)` or just, but I realize that would actually require even more effort ...

And now as I see that this demo doesn't even make sense, I'm starting to question that why would the server side item count event be fired when the component internal filtering is considered and the number of items doesn't change ? Because the semantics of the item count event (for all other components) are that it is only fired when the item count changes based on external filtering. I should have probably taken more thought about this even when I saw the confusing switch to server side filtering based on whether or not the listener is added.

One way to look at this would be to say that in the combo box case the item count change is never fired for internal filtering changes - since after the filtering is done and the user selects an item, the count reverts back to what it was anyway. So basically the real item count doesn't even change, just how much the user sees at the moment has changed. If anyone would need the number of items the CB is showing, it should be another event `FilteredItemCount` or such, but I don't know if that is even ever needed!

Reply:
Yes, that makes sense! I will revert these changes then, and make the data communicator to recognise whether it should fire the event or not.

Reply:
Disabling the firing an item count change event on the server side while applying a client filter is not a trivial change. This is closely connected with `ListDataView` API (add/remove/getItems/addFilter). The main problem is how to simultaneously disable event firing for client filter and enable for data view API. This will covered by a separate ticket.